### PR TITLE
[Minor] Added refresh support to measurement client

### PR DESCRIPTION
**Pull request by MounikaBattu17**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md). _(Required)_
- [ ]  Automatically post PR comments with images for G code changes? _(Recommended for small changes)_

### What does this Pull Request accomplish?
Added refresh support to the measurement client. This button updates the measurements list, resets the selection, and reverts any name changes in the `Available Measurements` list box.

### Why should this Pull Request be merged?
- Added a Refresh button in `Create Measurement Client.vi`.
- Updated the available services using `Enumerate Services.vi`.
- Utilized `List Available Measurements.vi` to refresh the list box with available services, resetting the selection and measurement service/client names.
- Disabled the `Create Measurement Client` button and set the visibility of `Warning` to false.

### What testing has been done?
Manually tested the functionality.

### CODE REVIEW

**Comment by MounikaBattu17** [comment]
Diff changes:
image
image

Refresh support

**Review by Avinash2Suresh — COMMENTED** [review]
When clicking the refresh button, it doesn’t provide the tactile feedback of a button click. Please consider updating the button to give users a sense of interaction, similar to how the Create Measurement Plug-in Client(s) button behaves.

**Comment by Avinash2Suresh** [comment]
When clicking the refresh button, it doesn’t provide the tactile feedback of a button click. Please consider updating the button to give users a sense of interaction, similar to how the Create Measurement Plug-in Client(s) button behaves.

File: Source/Generator/Create Measurement Client.vi
Comment (Avinash2Suresh) [inline Source/Generator/Create Measurement Client.vi:1]: When clicking the refresh button, it doesn’t provide the tactile feedback of a button click. Please consider updating the button to give users a sense of interaction, similar to how the Create Measurement Plug-in Client(s) button behaves.

Comment (Avinash2Suresh) [inline Source/Generator/Create Measurement Client.vi:1]: image

I see a refresh lock and release here. I believe we should do the same for the Refresh value change event.

**Review by Avinash2Suresh — CHANGES_REQUESTED** [review]
I would like to see the updated changes.

File: Source/Generator/Create Measurement Client.vi
Comment (dixonjoel) [reply Source/Generator/Create Measurement Client.vi:1]: Agree with this. I can't vouch for this web page, but there are ways to put an image onto a button.
https://forums.ni.com/t5/LabVIEW/picture-on-button/td-p/852758

Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: @dixonjoel 
Thank you for the resource! I created a custom refresh button based on the link. Could you please confirm if this button is okay?
image
Post your confirmation, I'll update the code.

Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: Here below is a small gif on the usage of button when running the client
Refresh support

Comment (dixonjoel) [reply Source/Generator/Create Measurement Client.vi:1]: Yeah, looks fine to me.

**Review by dixonjoel — APPROVED** [review]

File: Source/Generator/Create Measurement Client.vi
Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: Sure Thank you. I have updated the code with this new refresh button.

**Review by Avinash2Suresh — APPROVED** [review]

**Comment by Avinash2Suresh** [comment]
Please provide a tooltip for the Refresh button.

File: Source/Generator/Create Measurement Client.vi
Comment (Avinash2Suresh) [inline Source/Generator/Create Measurement Client.vi:1]: Consider adding a tooltip for the Refresh button.

Comment (Avinash2Suresh) [inline Source/Generator/Create Measurement Client.vi:1]: I would recommend to create a typedef for the Refresh button.

**Review by Prem-MysteryPro — APPROVED** [review]
Can we add tool tip to the refresh button? I guess here the only reason could refreshing the measurement list but still. Let us have it.
If we customized the icon for this button, can we save it as typedef?

File: Source/Generator/Create Measurement Client.vi
Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: image

Comment (MounikaBattu17) [reply Source/Generator/Create Measurement Client.vi:1]: Created a typedef `Refresh Button.ctl`.

**Pull request merged**